### ISSUE TITLE
Loss is nan use my data with CommonLanguage recipe

### ISSUE THREAD

**Issue by lhanzl** [speechbrain/speechbrain#898]
Use CommonLanguage recipes, replace my own data, and do a binary classification task. Why is loss is nan reported?

**Comment by TParcollet** [comment]
Hi, if it's with your own data, it might be because one of the sample is weird (empty, with frequency masking etc).

**Comment by mravanelli** [comment]
Sometimes that recipe can be numerically unstable. Make sure you use gradient clipping (add a parameter called --max_grad_norm=5 in the yaml file). Also make sure the learning rate is not too high.

**Comment by mravanelli** [comment]
Any update on  that? That recipe is sometimes numerically unstable and many times things are even solved by changing the seed (that will change the model initialization as well).

**Comment by lhanzl** [comment]
@TParcollet @mravanelli Thanks for your reply.
Sorry for not replying to the message in time.
I am sure that my data has no empty, sampling rate and other strange issues.
In fact, I have two environments. The biggest difference between them is that the GPU is different, and the same data will not appear nan when running in environment 1. But in another environment, no matter if you run the recipe directly or run the recipe with my data, I will encounter the nan problem, and the seed and max_grad_norm you mentioned cannot completely solve the problem.
I guess it may be an environmental problem, right?

**Comment by TParcollet** [comment]
Oof, this is ultra weird, but we already saw these kind of issues in the past ... You can try to degenerate your working environment to the second one until it doesn't work anymore ...
